### PR TITLE
add logs to document failing scenarios in kubemacpool startup.

### PR DESCRIPTION
**Pull request by RamLavi**
Signed-off-by: Ram Lavi <ralavi@redhat.com>

**What this PR does / why we need it**:
Currently in the main Run() method, we don't log error in case something fails.
We want to exit the Run() method should a irreconcilable error occur, and a proper log to be added in that case.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by alonSadan** [comment]
lgtm

**Comment by phoracek** [comment]
```
cmd/manager/main.go:25:2: imported and not used: "github.com/pkg/errors"
```

File: pkg/manager/manager.go [hunk 1]
```diff
@@ -92,13 +91,13 @@ func (k *KubeMacPoolManager) Run(rangeStart, rangeEnd net.HardwareAddr) error {
 
 		err = kubevirt_api.AddToScheme(k.runtimeManager.GetScheme())
 		if err != nil {
-			errors.Wrap(err, "unable to register kubevirt scheme")
+			return errors.Wrap(err, "unable to register kubevirt scheme")
```
Comment (qinqon) [inline pkg/manager/manager.go:94]: This changes behaviour and it will end `Run` are we sure we want that ?

File: pkg/manager/manager.go [hunk 2]
```diff
@@ -92,13 +91,13 @@ func (k *KubeMacPoolManager) Run(rangeStart, rangeEnd net.HardwareAddr) error {
...
 		}
 
 		isKubevirtInstalled := checkForKubevirt(k.clientset)
 		poolManager, err := poolmanager.NewPoolManager(k.clientset, rangeStart, rangeEnd, k.podNamespace, isKubevirtInstalled, k.waitingTime)
 		if err != nil {
-			errors.Wrap(err, "unable to create pool manager")
+			return errors.Wrap(err, "unable to create pool manager")
```
Comment (qinqon) [inline pkg/manager/manager.go:100]: ditto

File: pkg/manager/manager.go [hunk 3]
```diff
@@ -112,18 +111,18 @@ func (k *KubeMacPoolManager) Run(rangeStart, rangeEnd net.HardwareAddr) error {
 		log.Info("Setting up controllers")
 		err = controller.AddToManager(k.runtimeManager, poolManager)
 		if err != nil {
-			errors.Wrap(err, "unable to register controllers to the manager")
+			return errors.Wrap(err, "unable to register controllers to the manager")
```
Comment (qinqon) [inline pkg/manager/manager.go:114]: ditto

File: pkg/manager/manager.go [hunk 4]
```diff
@@ -112,18 +111,18 @@ func (k *KubeMacPoolManager) Run(rangeStart, rangeEnd net.HardwareAddr) error {
...
 		}
 
 		log.Info("Setting up webhooks")
 		err = webhook.AddToManager(k.runtimeManager, poolManager)
 		if err != nil {
-			errors.Wrap(err, "unable to register webhooks to the manager")
+			return errors.Wrap(err, "unable to register webhooks to the manager")
```
Comment (qinqon) [inline pkg/manager/manager.go:120]: ditto

File: pkg/manager/manager.go [hunk 5]
```diff
@@ -112,18 +111,18 @@ func (k *KubeMacPoolManager) Run(rangeStart, rangeEnd net.HardwareAddr) error {
...
 		}
 
 		err = k.runtimeManager.Start(k.restartChannel)
 		if err != nil {
-			errors.Wrap(err, "unable to run the manager")
+			log.Error(err, "unable to run the manager")
```
Comment (qinqon) [inline pkg/manager/manager.go:125]: Why don't we want to return here ?

File: pkg/manager/manager.go (hunk 1 above)
Comment (RamLavi) [reply pkg/manager/manager.go:94]: it does change behavior, but I think the former behavior was a bug. 
I mean, how can kubemacpool work properly if kubevirt scheme is not added? or if pool is not allocated correctly?

still it's a change so let's consult with @phoracek

File: pkg/manager/manager.go (hunk 5 above)
Comment (RamLavi) [reply pkg/manager/manager.go:125]: cases such as kubevirt was just applied and we want to restart the manager without restarting the pod

**Pull request merged**